src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java

```diff
@@ -187,6 +189,16 @@ public BitbucketCloudApiClient(boolean enableCache, int teamCacheDuration, int r
 
         // Create Http client
         HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
+
+        // https://hc.apache.org/httpcomponents-client-4.5.x/tutorial/html/connmgmt.html
+        ConnectionKeepAliveStrategy five_seconds_max = new ConnectionKeepAliveStrategy() {
+            @Override
+            public long getKeepAliveDuration(HttpResponse httpResponse, org.apache.http.protocol.HttpContext httpContext) {
```

Comment:
Could be written as a lambda 😓
Also, do we know what the current keep-alive header says?

Reply:
if you rewrite to a lambda you also avoid some of the imports.

Which fails in CI: https://travis-ci.org/jenkinsci/bitbucket-branch-source-plugin/builds/575015558#L312

Reply:
There is no keep-alive header produced by api.bitbucket.org, implying infinite.

Reply:
So I guess that is the problem 😆

Reply:
Other than this I haven't written any java since 1.6 and don't know how java lambdas work (in 1.8?). I can smash something together from a tutorial & stackoverflow... :/

Reply:
Ah, something like this

```java
httpClientBuilder.setKeepAliveStrategy((__, ___) -> {return MILLISECONDS.convert(5, SECONDS);});
```